src/Compilers/CSharp/Test/Emit3/FieldKeywordTests.cs

```diff
@@ -4752,6 +4752,1617 @@
+                //     public string? Prop { get; set => field = value; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableBackingField, "Prop").WithArguments("property", "Prop").WithLocation(7, 20));
+        }
+
+        [Fact]
+        public void Nullability_20_ManualGetter_AutoSetter()
+        {
+            // NotNull on field with manual getter and auto-setter
+            var source = """
+                #nullable enable
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    [field: NotNull]
+                    public string? Prop { get => field; set; }
+                }
+                """;
+
+            var comp = CreateCompilation([source, NotNullAttributeDefinition]);
+            comp.VerifyEmitDiagnostics(
+                // (7,20): warning CS9264: Non-nullable property 'Prop' must contain a non-null value when exiting constructor. Consider adding the 'required' modifier, or declaring the property as nullable, or adding '[field: MaybeNull, AllowNull]' attributes.
+                //     public string? Prop { get; set => field = value; }
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableBackingField, "Prop").WithArguments("property", "Prop").WithLocation(7, 20));
+        }
+
+        [Fact]
+        public void Nullability_20_ManualGetter_AutoSetter_NullInitializer()
+        {
+            // NotNull on field with manual getter and auto-setter and null initializer
+            // No diagnostic is expected here as the field's nullability+attrs is equivalent to `[AllowNull] string`--so, assigning null to it actually puts it into a non-null state.
+            var source = """
+                #nullable enable
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    [field: NotNull]
+                    public string? Prop { get => field; set; } = null;
+                }
+                """;
+
+            var comp = CreateCompilation([source, NotNullAttributeDefinition]);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullability_20_ManualGetter_AutoSetter_NotNullInitializer()
+        {
+            // NotNull on field with manual getter and auto-setter and non-null initializer
+            var source = """
+                #nullable enable
+                using System.Diagnostics.CodeAnalysis;
+
+                class C
+                {
+                    [field: NotNull]
+                    public string? Prop { get => field; set; } = "a";
+                }
+                """;
+
+            var comp = CreateCompilation([source, NotNullAttributeDefinition]);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void Nullability_20_ManualGetter_AutoSetter_NullTest()
```

Comment:
>Nullability_20_ManualGetter_AutoSetter_NullTest [](http://example.com/codeflow?start=20&length=47)

Is this distinct from the previous test? (What are we testing with the addition of `if (Prop is null) throw null!`?)

Reply:
The goal is to demonstrate that a null-test+throw on the property puts a non-nullable field into the correct state. Adding a version without initializer to demonstrate that more clearly.